game/scripts/vscripts/Sections/Section1.ts

```diff
@@ -0,0 +1,75 @@
+import * as tg from "../TutorialGraph/index"
+import * as tut from "../Tutorial/Core"
+import { getPlayerHero } from "../util"
+
+let graph: tg.TutorialStep | undefined = undefined
+let graphContext: tg.TutorialContext
+
+enum Section01EntityKeys {
+    CustomMudGolem = "npc_custom_mud_golem",
+    MudGolemTop = "mud_golem_top",
+    MudGolemBot = "mud_golem_bot"
+}
+
+const onStart = (complete: () => void) => {
+    print("Starting", "Section01");
+    CustomGameEventManager.Send_ServerToAllClients("section_started", { section: SectionName.Section01 })
+
+    const playerHero = getPlayerHero();
+    if (!playerHero) error("Could not find the player's hero.");
+
+    let mudGolemMeetPosition = playerHero.GetAbsOrigin().__add(Vector(300, 800, 0))
+
+    graph = tg.seq(
+        tg.setUnitMoveCapability(() => playerHero, UnitMoveCapability.NONE),
+        tg.setCameraTarget(() => playerHero),
+        tg.spawnAndMoveUnit(Section01EntityKeys.CustomMudGolem, 
+            playerHero.GetAbsOrigin().__add(Vector(0, 1500, 0)), 
+            mudGolemMeetPosition,
+            DotaTeam.GOODGUYS,
+            Section01EntityKeys.MudGolemTop),
+        tg.spawnAndMoveUnit(Section01EntityKeys.CustomMudGolem, 
+            playerHero.GetAbsOrigin().__add(Vector(1500, 500, 0)),
+            mudGolemMeetPosition.__add(Vector(150, -150, 0)), 
+            DotaTeam.GOODGUYS,
+            Section01EntityKeys.MudGolemBot),
+        tg.setCameraTarget(() => graphContext[Section01EntityKeys.MudGolemTop]),
+        tg.wait(3),
+        tg.setCameraTarget(() => graphContext[Section01EntityKeys.MudGolemBot]),
+        tg.faceTowards(Section01EntityKeys.MudGolemTop, playerHero.GetAbsOrigin()),
+        tg.faceTowards(Section01EntityKeys.MudGolemBot, playerHero.GetAbsOrigin()),
+        tg.wait(2),
+        tg.setCameraTarget(() => Entities.FindAllByName("dota_badguys_fort")[0]),
+        tg.wait(5),
+        tg.setCameraTarget(() => playerHero),
+    )
+
+    graphContext = {}
+
+    graph.start(graphContext, () => {
+        print("Completed", "Section01")
+        complete()
+    })
+}
+
+const onSkipTo = () => {
+    print("Skipping to", "Section01");
+    if (!getPlayerHero()) error("Could not find the player's hero.");
+
+    if (graphContext[Section01EntityKeys.MudGolemTop])
+        graphContext[Section01EntityKeys.MudGolemTop].RemoveSelf()
```

Comment:
Should removing the golems be in stop? Why remove them when skipping to this section, presumably we are in another section and they do not exist yet (and even if we are in this section at the moment it should call stop first).

Reply:
I believe we will have the mud golems in later sections so they have to be removed onSkip

Reply:
> I believe we will have the mud golems in later sections (maybe they could be cleared in the later sections?)

On second thought, I don't think it should be cleared there since there might be several sections with the golems and these particular ones are only spawned in this opening section

Reply:
Sunsfan gets killed in one of the first parts, idk if we respawn him later, so we would need to make sure we have the correct ones alive/dead in each section

Reply:
Found out this isn't even working properly, so I'll look into it a bit more tomorrow

Reply:
It seems both mudgolems die by the end of Script 1 and slacks and sunsfan become ghosts. Now I think for this section, calling clearMudGolems on both onSkipTo and onStop is sufficient. Added in this commit: https://github.com/ModDota/dota-tutorial/pull/50/commits/9d4c2c675f5c0fea67d6577ae81823c54c8b8a6d